src/CohortManager.js

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-vars */
+const Cohort = require('./Cohort')
+const Student = require('./Student')
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.students = []
+    this.studentsId = 1
+  }
+
+  addCohort(name) {
+    const regex = /^[A-Za-z0-9 ]*$/
+    if (name === '') {
+      return 'invalid name'
```

Comment:
Instead of returning a string, which can be cumbersome for whoever uses this createCohort function, an improvement here would be to throw an error: `throw new Error('A Cohort must have a name')`

To test that your function throws an error, your jasmine test could look like this:

```js
it('should throw an error when creating a cohort without a name', () => {
  expect(() => cManager.createCohort(null)).toThrow()
})
```

Notice that I've placed the createCohort call into a callback function when putting it into the expect - under the hood, the expect will run that callback function inside a try / catch block. It allows you to test for errors being thrown without crashing the tests